### PR TITLE
Empty bot generator

### PR DESCRIPTION
**Pull request by axelsrz**

### CODE REVIEW

**Comment by daveta** [comment]
> | [Send and receive messages](https://docs.microsoft.com/azure/bot-service/bot-builder-howto-send-messages?view=azure-bot-service-4.0&tabs=javascript) | The primary way your bot will communicate with users, and likewise receive communication, is through message activities. Some messages may simply consist of plain text, while others may contain richer content such as cards or attachments. |

Hmm - kind of brings up an interesting question.  Will we have Python tabs?

---
Refers to: generators/README.md:40 in 8840c9c. [](commit_id = 8840c9c4b68f9cfbe6821e60ddb4fdc535797c65, deletion_comment = False)

**Comment by daveta** [comment]
> | [Proactive messaging](https://docs.microsoft.com/en-us/azure/bot-service/bot-builder-howto-proactive-message?view=azure-bot-service-4.0) using [Adaptive Cards](https://docs.microsoft.com/azure/bot-service/bot-builder-send-welcome-message?view=azure-bot-service-4.0?#using-adaptive-card-greeting) | The primary goal when creating any bot is to engage your user in a meaningful conversation. One of the best ways to achieve this goal is to ensure that from the moment a user first connects to your bot, they understand your bot’s main purpose and capabilities.  We refer to this as "welcoming the user."  The Core template uses an [Adaptive Card](http://adaptivecards.io) to implement this behavior.  |

You can remove the en-us, it should do the right thing..

---
Refers to: generators/README.md:41 in 8840c9c. [](commit_id = 8840c9c4b68f9cfbe6821e60ddb4fdc535797c65, deletion_comment = False)

File: generators/app/templates/empty/{{cookiecutter.bot_name}}/config.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+class DefaultConfig:
+    """ Bot Configuration """
+
+    PORT = 3978
+    APP_ID = os.environ.get("MicrosoftAppId", "")
+    APP_PASSWORD = os.environ.get("MicrosoftAppPassword", "")
+    LUIS_APP_ID = os.environ.get("LuisAppId", "")
+    LUIS_API_KEY = os.environ.get("LuisAPIKey", "")
+    # LUIS endpoint host name, ie "westus.api.cognitive.microsoft.com"
+    LUIS_API_HOST_NAME = os.environ.get("LuisAPIHostName", "")
```
Comment (daveta) [inline generators/app/templates/empty/{{cookiecutter.bot_name}}/config.py]: Are these required for this bot?

File: generators/app/templates/empty/{{cookiecutter.bot_name}}/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+{{cookiecutter.bot_description}}
+
+This bot has been created using [Bot Framework](https://dev.botframework.com), it shows how to create a simple bot that accepts input from the user and echoes it back.
+
+## Prerequisites
+
+This sample **requires** prerequisites in order to run.
+
+### Install Python 3.7
```
Comment (daveta) [inline generators/app/templates/empty/{{cookiecutter.bot_name}}/README.md]: >3.7 [](start = 19, length = 3)

Are we 3.6 or 3.7?

File: generators/app/templates/empty/{{cookiecutter.bot_name}}/app.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    print(f"\n [on_turn_error] unhandled error: {error}", file=sys.stderr)
+
+    # Send a message to the user
+    await context.send_activity("The bot encounted an error or bug.")
+    await context.send_activity("To continue to run this bot, please fix the bot source code.")
+
+    # pylint: disable=protected-access
+    if hasattr(self, "_conversation_state"):
+        # If state was defined, cleare it.
+        await self._conversation_state.delete(context)
```
Comment (daveta) [inline generators/app/templates/empty/{{cookiecutter.bot_name}}/app.py]: cleare = clear
Do we need this in general? I don't see any usage of state..

File: generators/app/templates/empty/{{cookiecutter.bot_name}}/config.py (hunk 1 above)
Comment (axelsrz) [reply generators/app/templates/empty/{{cookiecutter.bot_name}}/config.py]: solved

File: generators/app/templates/empty/{{cookiecutter.bot_name}}/README.md (hunk 2 above)
Comment (axelsrz) [reply generators/app/templates/empty/{{cookiecutter.bot_name}}/README.md]: Solved

File: generators/app/templates/empty/{{cookiecutter.bot_name}}/app.py (hunk 3 above)
Comment (axelsrz) [reply generators/app/templates/empty/{{cookiecutter.bot_name}}/app.py]: Solved

**Comment by axelsrz** [comment]
> Small points, I think we should address the luis stuff in config and the phantom conversation_state cleanup in app.py, other than that fix at your discretion.

Took those changes.

**Pull request merged**